spdmlib/src/message/version.rs

```diff
@@ -100,19 +133,22 @@ impl SpdmCodec for SpdmVersionResponsePayload {
         }
 
         let mut versions = gen_array_clone(
-            SpdmVersionStruct {
-                update: 0,
-                version: SpdmVersion::SpdmVersion10,
-            },
+            SpdmVersionStruct::init(SpdmVersion::SpdmVersion10, 0),
             MAX_SPDM_VERSION_COUNT,
         );
 
         let mut version_count = 0;
         let rest = r.take(version_number_entry_count as usize * 2)?;
 
-        for i in 0..version_number_entry_count {
+        'version_read_loop: for i in 0..version_number_entry_count {
             if let Some(ver) = SpdmVersionStruct::read_bytes(&rest[i as usize * 2..]) {
                 if version_count < MAX_SPDM_VERSION_COUNT {
+                    for j in 0..i {
+                        if versions[j as usize].version == ver.version {
+                            // duplicated version, just ignore it
+                            continue 'version_read_loop;
+                        }
+                    }
```

Comment:
why need this?

Reply:
Will track this topic with another issue.